packages/grid/x-data-grid/src/hooks/features/rows/useGridParamsApi.ts

```diff
@@ -154,6 +160,38 @@ export function useGridParamsApi(apiRef: React.MutableRefObject<GridPrivateApiCo
     [apiRef, getBaseCellParams],
   );
 
+  const getRowValue = React.useCallback<GridParamsApi['getRowValue']>(
+    (row, colDef) => {
+      const id = getRowId ? getRowId(row) : row.id;
+      const field = colDef.field;
+
+      if (!colDef || !colDef.valueGetter) {
+        return row[field];
+      }
+
+      return colDef.valueGetter(getBaseCellParams(id, field));
+    },
+    [apiRef, getBaseCellParams, getRowId],
+  );
+
+  const getRowFormattedValue = React.useCallback<GridParamsApi['getRowFormattedValue']>(
+    (row, colDef) => {
+      const value = getRowValue(row, colDef);
+
+      if (!colDef || !colDef.valueFormatter) {
+        return value;
+      }
+
+      return colDef.valueFormatter({
+        id: getRowId ? getRowId(row) : row.id,
+        field: colDef.field,
+        value: value,
+        api: apiRef.current,
+      });
+    },
+    [apiRef, getBaseCellParams, getRowId, getRowValue],
+  );
+
```

Comment:
These functions allow to retrieve a row value directly. Indirection through `dataRowIdToModelLookup` is expensive.